### PR TITLE
Hud10837/manage operational layers kotlin

### PR DESCRIPTION
**Pull request by hud10837**
Hi @TADraeseke, this is the new kotlin implementation of manage operational layers. I rewrote it basically from scratch to remove the java version's extra classes and drag and drop logic, but in order to keep that functionality I've still put in quite a bit of UI code. To mitigate this I put the main logic of the sample up front in MainActivity and called the add/remove/reorder functions from the UI code so users don't need to worry about the UI. Could you take a look when you get the chance? Thanks!

### CODE REVIEW

File: kotlin/manage-operational-layers/build.gradle [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    // arcgis-android & appcompat lib dependencies from rootProject build.gradle
+    implementation "androidx.constraintlayout:constraintlayout:$constraintLayoutVersion"
+    implementation "com.google.android.material:material:$materialVersion"
+    implementation 'androidx.recyclerview:recyclerview:1.1.0'
```
Comment (TADraeseke) [inline kotlin/manage-operational-layers/build.gradle:28]: a recycler view version should be defined in version.gradle now

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import com.esri.arcgisruntime.layers.Layer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import kotlinx.android.synthetic.main.activity_main.*
+
+
+class MainActivity : AppCompatActivity() {
+
+  private val removedLayers = mutableListOf<Layer>()
```
Comment (TADraeseke) [inline kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt:37]: maybe this should match the naming in the UI, inactiveLayers?

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    val imageLayerCensus =
+      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")
+    val imageLayerDamage =
+      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/DamageAssessment/MapServer")
+
+    // add the map image layers to the map's operational layers
+    map.operationalLayers.addAll(listOf(imageLayerElevation, imageLayerCensus, imageLayerDamage))
+
+    // set the map to the map view
+    mapView.map = map
```
Comment (TADraeseke) [inline kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt:57]:
```suggestion
    val imageLayerElevation =
    ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer")
    val imageLayerCensus =
      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")
    val imageLayerDamage =
      ArcGISMapImageLayer("https://sampleserver5.arcgisonline.com/arcgis/rest/services/DamageAssessment/MapServer")

    // create a map with a topographic basemap and set it to the map view
    mapView.map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 34.056295, -117.195800, 4).apply {
      // add the map image layers to the map's operational layers
      operationalLayers.addAll(listOf(imageLayerElevation, imageLayerCensus, imageLayerDamage))
    }
```

Maybe I'm going overboard with inlining?

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+   */
+  private fun removeLayerFromMap(position: Int) {
+    val operationalLayers = mapView.map.operationalLayers
+    // store the layer in a list to keep track of which layers have been removed
+    removedLayers.add(operationalLayers[position])
+    // remove the layer from the map
+    operationalLayers.removeAt(position)
+
+    // tell the recycler view that the item has been removed
+    activeRecyclerView.adapter?.notifyDataSetChanged()
```
Comment (TADraeseke) [inline kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt:93]: Can this be activeRecyclerView.adapter?.notifyItemRemoved(position) ?

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+   */
+  private fun addLayerToMap(position: Int) {
+    // remove the layer from the removed layers
+    val layer = removedLayers.removeAt(position)
+    // add the layer to the map
+    mapView.map.operationalLayers.add(layer)
+
+    // notify the recycler views of the change
+    removedRecyclerView.adapter?.notifyDataSetChanged()
+    activeRecyclerView.adapter?.notifyDataSetChanged()
```
Comment (TADraeseke) [inline kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt:109]: I guess these could be:

    removedRecyclerView.adapter?.notifyItemRemoved(position)
    activeRecyclerView.adapter?.notifyItemInserted(mapView.map.operationalLayers.size)

but maybe makes it more confusing?

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt (hunk 3 above)
Comment (hud10837) [reply kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt:57]: Hm, I'm torn, this one is reasonable but it also seems like it's a lot of operations. Maybe we should try to define when we inline stuff in general? I can go either way here

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt (hunk 2 above)
Comment (hud10837) [reply kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt]: yes, I meant to do this. I've also renamed the view id and the adapter class

File: kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt (hunk 5 above)
Comment (hud10837) [reply kotlin/manage-operational-layers/src/main/java/com/esri/arcgisruntime/sample/manageoperationallayers/MainActivity.kt]: I think this is clearer for both of the above! I wasn't aware of these methods

File: kotlin/manage-operational-layers/build.gradle (hunk 1 above)
Comment (hud10837) [reply kotlin/manage-operational-layers/build.gradle:28]: ooh, fancy!

**Comment by hud10837** [comment]
@TADraeseke thanks, I've made all of those changes now!

**Comment by hud10837** [comment]
Thanks! Merging now

**Pull request merged**